blenderproc/python/loader/URDFLoader.py

```diff
@@ -0,0 +1,454 @@
+    parent_link = one_by_attr(elements=links, attr_name="name", value=joint_tree.parent)
+
+    # determine full transformation matrix
+    mat = Matrix(parent_link.get_local2world_mat()) @ Matrix(joint_tree.origin)
+    child_link.set_local2world_mat(mat)
+    child_link.set_parent(parent=parent_link)
+    parent_link.set_child(child=child_link)
+
+    for obj in child_link.get_all_objs():
+        obj.set_local2world_mat(Matrix(child_link.get_local2world_mat()) @ Matrix(obj.get_local2world_mat()))
+
+    # set link2bone mat
+    if child_link.bone is not None:
+        child_link.set_link2bone_mat(mat.inverted() @ child_link.bone.matrix)
+
+    if recursive:
+        child_joint_trees = get_joints_which_have_link_as_parent(child_link.get_name(), joint_trees)
+        for child_joint_tree in child_joint_trees:
+            propagate_pose(links, child_joint_tree, joint_trees, armature, recursive=True)
+
+
+def load_geometry(geometry_tree: "urdfpy.Geometry", urdf_path: Union[str, None] = None) -> MeshObject:
+    """ Loads a geometric element from a urdf tree.
+
+    :param geometry_tree: The urdf representation of the geometric element.
+    :param urdf_path: Optional path of the urdf file for relative geometry files.
+    :return: The respective MeshObject.
+    """
+    if geometry_tree.mesh is not None:
+        if os.path.isfile(geometry_tree.mesh.filename):
+            obj = load_obj(filepath=geometry_tree.mesh.filename)[0]
+        elif urdf_path is not None and os.path.isfile(urdf_path):
+            relative_path = os.path.join('/'.join(urdf_path.split('/')[:-1]), geometry_tree.mesh.filename)
+            if os.path.isfile(relative_path):
+                # load in default coordinate system
+                obj = load_obj(filepath=relative_path, axis_forward='Y', axis_up='Z')[0]
+            else:
+                print(f"Couldn't load mesh file for {geometry_tree} (filename: {geometry_tree.mesh.filename}; urdf "
+                      f"filename: {urdf_path})")
+        else:
+            raise NotImplementedError(f"Couldn't load mesh file for {geometry_tree} (filename: "
+                                      f"{geometry_tree.mesh.filename})")
+    elif geometry_tree.box is not None:
+        obj = create_primitive(shape="CUBE")
+        obj.blender_obj.dimensions = Vector(geometry_tree.box.size)
+    elif geometry_tree.cylinder is not None:
+        obj = create_primitive(shape="CYLINDER", radius=geometry_tree.cylinder.radius,
+                               depth=geometry_tree.cylinder.length)
+    elif geometry_tree.sphere is not None:
+        obj = create_primitive(shape="SPHERE", radius=geometry_tree.sphere.radius)
+    else:
+        raise NotImplementedError(f"Unknown geometry in urdf_tree {geometry_tree}")
+    obj.persist_transformation_into_mesh(location=True, rotation=True, scale=True)
+    return obj
+
+
+def load_viscol(viscol_tree: Union["urdfpy.Visual", "urdfpy.Collision"], name: str,
```

Comment:
Do we really have to save so many letters here? `load_viscol` -> `load_visual_collision_obj`?

Remember someone else has to understand what this function does based on the name.